Wrappers/Python/cil/utilities/dataexample.py

```diff
@@ -645,4 +356,150 @@ def scikit_random_noise(image, mode='gaussian', seed=None, clip=True, **kwargs):
+    dfile = TestData.RESOLUTION_CHART
+class SIMPLE_PHANTOM_2D(TestData):
+    dfile = TestData.SIMPLE_PHANTOM_2D
+class SHAPES(TestData):
+    dfile = TestData.SHAPES
+class RAINBOW(TestData):
+    dfile = TestData.RAINBOW
+
+class NexusTestData(BaseTestData):
+    @classmethod
+    def get(cls, data_dir=DEFAULT_DATA_DIR):
+        '''
+        Returns
+        -------
+        AcquisitionData
+        '''
+        loader = NEXUSDataReader()
+        loader.set_up(file_name=os.path.join(data_dir, cls.dfile))
+        return loader.read()
+
+class SYNCHROTRON_PARALLEL_BEAM_DATA(NexusTestData):
+    '''A DLS dataset'''
+    dfile = '24737_fd_normalised.nxs'
+class SIMULATED_PARALLEL_BEAM_DATA(NexusTestData):
+    '''A simulated parallel-beam dataset generated from SIMULATED_SPHERE_VOLUME'''
+    dfile = 'sim_parallel_beam.nxs'
+class SIMULATED_CONE_BEAM_DATA(NexusTestData):
+    '''A cone-beam dataset generated from SIMULATED_SPHERE_VOLUME'''
+    dfile = 'sim_cone_beam.nxs'
+class SIMULATED_SPHERE_VOLUME(NexusTestData):
+    '''A simulated volume of spheres'''
+    dfile = 'sim_volume.nxs'
+
+class RemoteTestData(BaseTestData):
+    URL: str
+    FILE_SIZE: str
+
+    @staticmethod
+    def _prompt(msg):
+        while (res := input(f"{msg} [y/n]").lower()) not in "yn":
+            pass
+        return res == "y"
+
+    def download_data(self):
+        '''Download a dataset from a remote repository'''
+        folder = os.path.join(self.data_dir, type(self).__name__)
+        if os.path.isdir(folder):
+            print(f"Dataset already exists in {folder}")
+        else:
+            if self._prompt(f"Are you sure you want to download {self.FILE_SIZE} dataset from {self.URL}?"):
+                print(f"Downloading dataset from {self.URL}")
+                with urlopen(self.URL) as response, BytesIO(response.read()) as bytes, ZipFile(bytes) as zipfile:
+                    zipfile.extractall(path=folder)
+                print('Download complete')
+            else:
+                print('Download cancelled')
+
+class WALNUT(RemoteTestData):
+    '''A microcomputed tomography dataset of a walnut from https://zenodo.org/records/4822516'''
+    URL = 'https://zenodo.org/record/4822516/files/walnut.zip'
+    FILE_SIZE = '6.4 GB'
+
+    @classmethod
+    def get(cls, data_dir=DEFAULT_DATA_DIR):
+        '''
+        This function returns the raw projection data from the .txrm file
+
+        Returns
+        -------
+        ImageData
+            The walnut dataset
+        '''
+        self = cls(data_dir)
+        filepath = os.path.join(self.data_dir, cls.__name__, 'valnut','valnut_2014-03-21_643_28','tomo-A','valnut_tomo-A.txrm')
+        try:
+            loader = ZEISSDataReader(file_name=filepath)
+            return loader.read()
+        except FileNotFoundError as exc:
+            raise ValueError(f"Specify a different data_dir or download data with `{cls.__name__}({data_dir}).download_data()`") from exc
```

Comment:
btw @hrobarts why not this instead?

```suggestion
        self.download_data()
        loader = ZEISSDataReader(file_name=filepath)
        return loader.read()
```

Reply:
We had some discussion about wanting to keep the download separate from get so the user has to actively request the download. But I agree as we have the prompt before downloading this would be neater...

Reply:
also could have `def get(..., missing='prompt' / 'download' / 'error')` to control behaviour